system/filesystem.py

```diff
@@ -82,9 +136,18 @@ def main():
     rc,raw_fs,err = module.run_command("%s -c /dev/null -o value -s TYPE %s" % (cmd, dev))
     fs = raw_fs.strip()
 
-
-    if fs == fstype:
+    if fs == fstype and resizefs == False:
         module.exit_json(changed=False)
+    elif fs == fstype and resizefs == True:
+        cmd = module.get_bin_path(growcmd, required=True)
+        if module.check_mode:
+            changed = True
```

Comment:
shouldn't we just exit now with changed=True? Now, if we are in check mode, execution will continue till next  check for module.check_mode on line 156. This flow seems not very obvious and error-prone in future.